tfexec/internal/e2etest/graph_test.go

```diff
@@ -0,0 +1,29 @@
+package e2etest
+
+import (
+	"context"
+	"testing"
+
+	"github.com/hashicorp/go-version"
+
+	"github.com/hashicorp/terraform-exec/tfexec"
+)
+
+func TestGraph(t *testing.T) {
+	runTest(t, "basic", func(t *testing.T, tfv *version.Version, tf *tfexec.Terraform) {
+		err := tf.Init(context.Background())
+		if err != nil {
+			t.Fatalf("error running Init in test directory: %s", err)
+		}
+
+		err = tf.Apply(context.Background())
+		if err != nil {
+			t.Fatalf("error running Apply: %s", err)
+		}
+
+		_, err = tf.Graph(context.Background())
```

Comment:
I noticed minor differences in string value for certain resources in the graph output. Because of this i am validating against one known good string consistent in all outputs. I'm not sure if this satisfies your request or not.

Reply:
I think this is much better than before, thank you!

For context, this library is becoming one of the ways we smoke test new Terraform CLI releases, so the subtle differences is actually something we may want to know about going forward. However for now this will do. 👌🏻